### ISSUE TITLE
Pylance not working for wrapped types

### ISSUE THREAD

**Issue by AmevinLS** [NowanIlfideme/pydantic-cereal#56]
Pylance sees the wrapped types as `Any`, which disables hinting, autocompletion, etc.
This will be very annoying to users

**Comment by AmevinLS** [comment]
Currently there is one workaround that allows for correct type hinting
Example:
```
PandasDF = cereal.wrap_type(pd.DataFrame, reader=pd_read, writer=pd_write)
PandasDFAlias = PandasDF

pdf1: PandasDF = pd.DataFrame()
pdf2: PandasDFAlias = pd.DataFrame()
```
The `PandasDF` is treated as a `(variable)`, and therefore `pdf1` doesn't have Pylance completions/support
The `PandasDFAlias` is treated as a `(type alias)`, and therefore `pdf2` has Pylance completions/support

However this would be on the users to do, which is cumbersome/inconvenient and will often be forgotten about - overall not good.

Will investigate other options in hopes of not relying on this workaround

**Comment by NowanIlfideme** [comment]
Ideally we could just return a `TypeAlias` in the signature... https://stackoverflow.com/a/70783271
But actually the issue is deeper:

```python
from typing_extensions import TypeAlias

MyWrappedType: TypeAlias = cereal.wrap_type(MyType, reader=my_reader, writer=my_writer)
```

this gives:

```text
Invalid expression form for type alias definitionPylance[reportInvalidTypeForm](https://github.com/microsoft/pyright/blob/main/docs/configuration.md#reportInvalidTypeForm)
Call expression not allowed in type expressionPylance[reportInvalidTypeForm](https://github.com/microsoft/pyright/blob/main/docs/configuration.md#reportInvalidTypeForm)
```

**Comment by AmevinLS** [comment]
Yes, a _type alias_ definition only allows some simple assignment operation. And the function always returns a _variable_ (even if it evaluates to a _type_). So it's probably impossible to return a _type alias_ directly from the `wrap_type` method

**Comment by NowanIlfideme** [comment]
I have come to the conclusion that it's impossible to make a "valid" type form here... so I guess we should suggest the following?

```python
MyWrappedType: TypeAlias = cereal.wrap_type(MyType, ...)  # type: ignore
```

That worked for Pyright 😄 

Unfortunately, `MyWappedType: TypeAlias[MyType] = ...  # type: ignore` did NOT work for Pyright, as it broke downstream!

**Comment by AmevinLS** [comment]
I might've found a solution:

```python
from typing import NewType

MyWrappedType = NewType(
    "MyWrappedType", 
    cereal.wrap_type(MyType, ...)
)
```

This seems to allow the editor to parse type correctly and resolve the `Variable not allowed in type expression Pylance(reportInvalidTypeForm)` warning.

**Comment by NowanIlfideme** [comment]
Still really verbose, but I guess it works. 😄
